### PR TITLE
Dependency Version Bumps

### PR DESCRIPTION
**Pull request by lilleswing**
After the cut time to get on the latest libraries.

Python 3.6 has been out long enough to be stable and have all the libraries we need (3.7 was released last month!).  We should transition to 3.6 by default.

### CODE REVIEW

**Comment by lilleswing** [comment]
So I can't get the new eager paradigm to work with TensorGraph save/load.

Eager is now no longer compatible with graphs.  You cannot be in a Graph Context at the same time as an eager context.

When in an Eager Context our layers when Called with create new weights (this is what is desired).  These new weights will not have the same variable names as the first time they were created so the save/restore can't work.

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 1]
```diff
@@ -633,23 +648,22 @@ def build_layers(layer, tensors):
         return tensor
 
       tensors = {}
-      with self._get_tf("Graph").as_default():
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tensor_graph.py]: no longer allowed

File: deepchem/models/tensorgraph/tests/test_tensor_graph.py [hunk 2]
```diff
@@ -219,8 +217,26 @@ def test_set_optimizer(self):
   @flaky
   def test_set_optimizer_eager(self):
     with context.eager_mode():
-      with tfe.IsolateTest():
-        self.test_set_optimizer()
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tests/test_tensor_graph.py]: this test was passing except for the part which also tested saving and loading

**Comment by peastman** [comment]
> Eager is now no longer compatible with graphs.

I'll take a look at it.

**Comment by lilleswing** [comment]
> How about moving the eager related things into a separate PR so we can work those out first? Looking through the documentation I see a few other changes we should make.

> * Change tfe.enable_eager_execution() to tf.enable_eager_execution(). (I think this should only appear in the examples.)
> * Change tfe.in_eager_mode() to tf.executing_eagerly().


> You mentioned some problems with loading and saving. I don't see why there would be an issue with that. Saver supports eager mode, and we always pass a list of variables to its constructor, so we're not relying on a graph for that.

So the saver and loader do work.  The issue is how we construct the variables in TensorGraph.load_from_dir.

The way TensorGraph.load_from_dir works in graph mode is
1) It constructs a new "Graph" -- you can think of this as a clean workspace to drop variables onto.
2) It runs the layers.call() in topological sort order.  This CREATES new variables identified by NAME.  It automatically generates names for the variables based on how many variables existed in the "Graph" before.  That is if a variable is the third time a Dense.__call__() is called it will be called dense_3.
3) It restores from a checkpoint file.  Checkpoint files are essentially a dictionary of {variable_name:tensor}  It aligns the variables by name in the "Graph"

The issue now with eager mode is we can no longer do (1), creating a "clean" workspace.  Since we can't create a clean workspace (2) now generates variables but their names do not align.  Instead their names start where the workspace left off.  Then when we load from the checkpoint file the variable names don't align and the variables do not go into the right location.

**Comment by peastman** [comment]
It looks like they got rid of the ability to use Graphs in eager mode.  (Thanks Google!)  And in typical Tensorflow style, their response to a problem is to create a completely new, mutually incompatible tool with a similar but not identical API to the old one.  So now we're supposed to start using Checkpoints instead of Savers.

https://www.tensorflow.org/api_docs/python/tf/train/Checkpoint

**Comment by lilleswing** [comment]
Leave this as dead for now -- I handle many of the issues in TensorGraph load from json.  Once that is resolved i will attempt to bump TF again.

**Comment by rbharath** [comment]
@lilleswing Do you think it would be easier to bump the version now? We should starting getting up to latest TensorFlow in preparation for TF 2.0 bump

**Comment by rbharath** [comment]
I'm going to close this old PR since I think we've made all these version bumps already.

**Pull request closed without merging**